DataPlotly/ui/dataplotly_dockwidget_base.ui

```diff
@@ -770,119 +769,190 @@ QListWidget::item::selected {
+                   </property>
+                   <property name="checked">
                     <bool>true</bool>
                    </property>
+                  </widget>
+                 </item>
+                 <item row="16" column="0">
+                  <widget class="QLabel" name="bar_mode_lab">
                    <property name="text">
-                    <string>Invert Histogram Direction</string>
+                    <string>Bar Mode</string>
                    </property>
                   </widget>
                  </item>
-                 <item row="22" column="1">
-                  <widget class="QgsDoubleSpinBox" name="bar_gap">
-                   <property name="maximum">
-                    <double>1.000000000000000</double>
+                 <item row="25" column="0" colspan="3">
+                  <spacer name="verticalSpacer_6">
+                   <property name="orientation">
+                    <enum>Qt::Vertical</enum>
                    </property>
-                   <property name="singleStep">
-                    <double>0.100000000000000</double>
+                   <property name="sizeHint" stdset="0">
+                    <size>
+                     <width>255</width>
+                     <height>222</height>
+                    </size>
+                   </property>
+                  </spacer>
+                 </item>
+                 <item row="11" column="1">
+                  <widget class="QLabel" name="y_axis_mode_label">
+                   <property name="text">
+                    <string>Y Axis Mode</string>
                    </property>
                   </widget>
                  </item>
-                 <item row="18" column="0">
-                  <widget class="QLabel" name="violinSideLabel">
+                 <item row="10" column="1">
+                  <widget class="QLabel" name="x_axis_mode_label">
                    <property name="text">
-                    <string>Violin side</string>
+                    <string>X Axis Mode</string>
                    </property>
                   </widget>
                  </item>
-                 <item row="18" column="1" colspan="2">
-                  <widget class="QComboBox" name="violinSideCombo"/>
+                 <item row="12" column="0" colspan="3">
+                  <widget class="QgsCollapsibleGroupBox" name="x_axis_bounds_check">
+                   <property name="title">
+                    <string>Set X Axis bounds</string>
+                   </property>
+                   <property name="checkable">
+                    <bool>true</bool>
+                   </property>
+                   <property name="checked">
+                    <bool>false</bool>
+                   </property>
+                   <layout class="QGridLayout" name="gridLayout_11">
+                    <item row="0" column="1">
+                     <widget class="QLineEdit" name="x_axis_min"/>
```

Comment:
Maybe a QgsDoubleSpinBox would be more suitable for these controls?

Reply:
I have replaced the QLineEdits with QgsDoubleSpinBoxes.

Reply:
Hmm... Does it make sense to use QgsDoubleSpinBoxes? I have a use case where I want to set the bounds of a Date variable (e.g. 06:00 - 20:00)? I know, dates do not work right now (#122)

Reply:
Hm - good point, but maybe we should adapt in future to detect the field type and choose an appropriate widget to suit? (e.g. calendar widget for date fields)

Reply:
Sounds good